pandas/tests/arrays/string_/test_string.py

```diff
@@ -194,6 +194,24 @@ def test_constructor_raises():
     with pytest.raises(ValueError, match="sequence of strings"):
         pd.arrays.StringArray(np.array([]))
 
+    with pytest.raises(ValueError, match="strings or pandas.NA"):
+        pd.arrays.StringArray(np.array(["a", np.nan], dtype=object))
+
+    with pytest.raises(ValueError, match="strings or pandas.NA"):
+        pd.arrays.StringArray(np.array(["a", None], dtype=object))
+
+    with pytest.raises(ValueError, match="strings or pandas.NA"):
+        pd.arrays.StringArray(np.array(["a", pd.NaT], dtype=object))
+
+
+def test_from_sequnce_no_mutate():
+    a = np.array(["a", pd.NA], dtype=object)
```

Comment:
use np.nan or None here? Otherwise the original won't be different from the potentially mutated one?

Reply:
Whoops, yes. That's what I intended.